challenger_test.go

```diff
@@ -100,15 +100,18 @@ func newChallenger() (*LndChallenger, *mockInvoiceClient, chan error) {
 	}
 	invoicesMtx := &sync.Mutex{}
 	mainErrChan := make(chan error)
-	return &LndChallenger{
+	c := &LndChallenger{
 		Client:        mockClient,
 		GenInvoiceReq: genInvoiceReq,
 		invoiceStates: make(map[lntypes.Hash]lnrpc.Invoice_InvoiceState),
 		quit:          make(chan struct{}),
 		invoicesMtx:   invoicesMtx,
 		invoicesCond:  sync.NewCond(invoicesMtx),
 		errChan:       mainErrChan,
-	}, mockClient, mainErrChan
+	}
+	c.VerifyInvoiceStatusFunc = c.DefaultVerifyInvoiceStatus
```

Comment:
nit: can add directly to the struct initialization.

Reply:
I think there's a problem trying to assign it directly in the struct because it relies on an initialized challenge in `DefaultVerifyInvoiceStatus`. Wasn't sure how much of an anti pattern it was to do it the way I did or not but the side effect of it is that it can't be initialized directly.